### PR TITLE
Add user helper functions for loading env from k8s secrets

### LINKED ISSUE #366: Email notification support
/kind feature

**Description:**
[A clear and concise description of what your proposal. What problem does it solve?]
Tekton community already has sendmail catalog: https://github.com/tektoncd/catalog/tree/master/task/sendmail, kfp-tekton community can enhance dsl to support it accordingly.

### Details of the the notification requirement:

User can specify a parameter whose value you indicate at run time for the SMTP Mail server name field, Senders email address field, Recipients email address field, and Email subject field. 

The Notification stage includes the following fields.
* SMTP Mail server name
> The name of the server or its IP address. You can specify a parameter whose value you indicate at run time.
Senders email address
The email address that the notification is sent from.
Recipients email address
The email address that the notification is sent to. You can specify multiple email addresses, separated by a space.
* Email subject
> The text that is included in the subject line of the notification.
* Attachments
> Files to be sent with the notification. Specify a path name or a comma-separated list of pathnames, which must be enclosed in single-quotes or double-quotes. You can also specify an expression that resolves to a pathname or comma-separated pathnames.
* Email body
> The text that is included in the body of the notification.
* Include job status in email
> specifies whether you want to include available job status information in the message.

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #366

**Description of your changes:**
For some commonly used k8s resources, it's annoying to ask users to add their copy the example helper functions when they are creating their own custom pipeline. Thus, This PR set up the ground work and added the helper function for loading env from k8s secrets to begin with.

@jinchihe 

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: samples/exit-handler-email/send-email.py [hunk 1]
```diff
@@ -14,10 +14,9 @@
 
 from kfp import dsl
 import kfp
-from kubernetes import client as k8s_client
 from kfp.components import func_to_container_op
 from kfp import onprem
-import os
+from kfp_tekton.k8s_client_helper import env_from_secret
```
Comment (fenglixa) [inline samples/exit-handler-email/send-email.py:19]: I saw you defined the env_from_secret here, but I didn't see any caller in this script.

**Comment by fenglixa** [comment]
Thanks @Tomcli, good idea that handle the env via DSL apply function.
@jinchihe, could you please help review? Thanks.
/cc @jinchihe

File: samples/exit-handler-email/send-email.py (hunk 1 above)
Comment (Tomcli) [reply samples/exit-handler-email/send-email.py:19]: it's being used at line 51-55

**Comment by jinchihe** [comment]
Wow! Great feature! @Tomcli

File: samples/exit-handler-email/send-email.py (hunk 1 above)
Comment (jinchihe) [reply samples/exit-handler-email/send-email.py:19]: Yeah, that's stange, why the 51-55 are not shown in the diff.

File: sdk/python/kfp_tekton/k8s_client_helper.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2019-2020 kubeflow.org.
```
Comment (jinchihe) [inline sdk/python/kfp_tekton/k8s_client_helper.py:1]: 2020 -> 2021

**Review by jinchihe — COMMENTED** [review]
Looks good.

**Pull request merged**